### PR TITLE
implemented task

### PR DESCRIPTION
**Pull request by ihorvasylenko23**

### CODE REVIEW

File: src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (keda-vlad) [inline src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java]: U don't need @Autowired if u have only 1 constructor

File: src/main/java/mate/academy/SpringBootProjectApplication.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class SpringBootProjectApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (keda-vlad) [inline src/main/java/mate/academy/SpringBootProjectApplication.java]: Make injection here by constructor like in BookService

File: src/main/java/mate/academy/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private String author;
+
+    @Column(nullable = false, unique = true)
+    private String isbn;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    private String description;
+    private String coverImage;
```
Comment (keda-vlad) [inline src/main/java/mate/academy/model/Book.java:33]: https://medium.com/@centizennationwide/mysql-naming-conventions-e3a6f6219efe
use  @Column(name = "cover_image")

**Review by keda-vlad — CHANGES_REQUESTED** [review]

**Comment by ohatsenko** [comment]
I completely agree with Vlad

File: src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t insert book into DB: " + book);
```
Comment (ArtemReizer) [inline src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java]: if I'm not mistaken, throwing a simple RuntimeException is bad practice. You can throw HibernateException for example

**Review by ArtemReizer — CHANGES_REQUESTED** [review]
GJ!) You can improve it

File: src/main/java/mate/academy/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (ohatsenko) [inline src/main/java/mate/academy/model/Book.java]:
```suggestion
@Data
@Table(name = "books")
public class Book {

```

File: src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooksQuery = session.createQuery("from Book", Book.class);
+            return getAllBooksQuery.getResultList();
+        }
+    }
```
Comment (ohatsenko) [inline src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java:45]:
```suggestion
        }  catch (Exception e) {
            throw new EntityNotFoundException("Can`t get all books", e);
        }
    }

```

**Review by ohatsenko — CHANGES_REQUESTED** [review]

**Review by ohatsenko — APPROVED** [review]

File: checkstyle.xml [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        <module name="EmptyCatchBlock">
+            <property name="exceptionVariableName" value="expected"/>
+        </module>
+        <module name="CommentsIndentation"/>
+        <module name="DeclarationOrder"/>
+        <module name="SingleSpaceSeparator">
+            <property name="validateComments" value="true"/>
+        </module>
+    </module>
+</module>
```
Comment (vovasalyha) [inline checkstyle.xml]: Empty line

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+		</dependency>
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-orm</artifactId>
```
Comment (vovasalyha) [inline pom.xml]: All hibernate related dependencies are already bundled in `data-jpa` starter

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+			<version>6.0.11</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-dbcp2</artifactId>
```
Comment (vovasalyha) [inline pom.xml]: Where are you using it?

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-checkstyle-plugin</artifactId>
+				<version>3.1.1</version>
```
Comment (vovasalyha) [inline pom.xml]: Use the latest version of plugin

File: src/main/java/mate/academy/SpringBootProjectApplication.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        this.bookService = bookService;
+    }
+
+    public static void main(String[] args) {
+        SpringApplication.run(SpringBootProjectApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/SpringBootProjectApplication.java]: Use lambda

File: src/main/java/mate/academy/SpringBootProjectApplication.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class SpringBootProjectApplication {
+    private final BookService bookService;
+
+    public SpringBootProjectApplication(BookService bookService) {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/SpringBootProjectApplication.java]: You can use lombok annotation to generate constructor.

File: src/main/java/mate/academy/model/Book.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+@Table(name = "books")
```
Comment (vovasalyha) [inline src/main/java/mate/academy/model/Book.java]:
```suggestion
@Table(name = "book")
```

File: src/main/java/mate/academy/model/Book.java (hunk 3 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/model/Book.java:33]: It's not necessary, hibernate default resolving strategy automatically parsing it to snake_case

File: src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java]: This method is deprecated in 6 version.

File: src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooksQuery = session.createQuery("from Book", Book.class);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/repositiry/impl/BookRepositoryImpl.java]:
```suggestion
            Query<Book> getAllBooksQuery = session.createQuery("FROM Book", Book.class);

```

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve few things.

**Review by boroda4436 — APPROVED** [review]
LGTM

**Pull request merged**